src/main/kotlin/baseball/model/NumberComparator.kt

```diff
@@ -0,0 +1,21 @@
+package baseball.model
+
+import baseball.config.GameNumberConfig
+
+class NumberComparator {
+
+    fun compare(input: String, answer: List<Int>): BallAndStrike {
+        val ballAndStrike = BallAndStrike()
+        for (i in 0 until GameNumberConfig.DIGIT_NUMBER) {
+            if (answer.contains(input[i] - '0')) {
+                ballAndStrike.ball++
+            }
+            if (input[i] - '0' == answer[i]) {
+                ballAndStrike.strike++
+            }
+        }
+        ballAndStrike.ball -= ballAndStrike.strike
+        return ballAndStrike
+    }
```

Comment:
strike와 ball if문의 순서를 바꾸면 연산 횟수를 줄일 수 있습니다.

```suggestion
    fun compare(input: String, answer: List<Int>): BallAndStrike {
        val ballAndStrike = BallAndStrike()
        for (i in 0 until GameNumberConfig.DIGIT_NUMBER) {
            if (input[i] - '0' == answer[i]) {
                ballAndStrike.strike++
                continue
            }
            if (answer.contains(input[i] - '0')) {
                ballAndStrike.ball++
            }
        }
        return ballAndStrike
    }
```

Reply:
strike과 ball의 연산이 각각 O(1), O(n) 인 것 같아 스트라이크 계산이 앞에 오는 게 용이하겠네요!
저도 앞으로 이 부분은 신경 쓰면서 코딩해야겠습니다.

Reply:
오 이 부분 생각지 못했는데 이런 부분도 신경써야 겠군요 감사합니다!!